lib/engine/route.rb

```diff
@@ -44,7 +44,7 @@ def next_connection(node, connection, other)
     end
 
     def select(node, other)
-      other_paths = @routes.reject { |r| r == self }.flat_map(&:paths)
+      other_paths = @routes.flat_map(&:paths)
```

Comment:
so why is this one changed?

Reply:
Because otherwise clicking on a node can re-use a path on the current route. This was the first issue I encountered when I prototyped parallel paths.